django/db/models/expressions.py

```diff
@@ -360,28 +371,27 @@ def flatten(self):
             if expr:
                 yield from expr.flatten()
 
+    @cached_property
+    def identity(self):
+        constructor_signature = inspect.signature(self.__init__)
+        args, kwargs = self._constructor_args
+        signature = constructor_signature.bind_partial(*args, **kwargs)
```

Comment:
`bind_partial` is required to avoid passing `self` and ignoring it later on. If we've it to this point it's safe to assume we the object went through a successful initialization process.